### PR TITLE
Fix a few minor Lean misformalisations

### PR DESCRIPTION
**Pull request by ocfnash**

### CODE REVIEW

File: lean4/src/putnam_1994_b3.lean [hunk 1]
```diff
@@ -9,8 +9,7 @@ abbrev putnam_1994_b3_solution : Set ℝ := sorry
...
 -/
 theorem putnam_1994_b3
-(k : ℝ)
-(allfexN : Prop)
-(hallfexN : allfexN = ∀ f : ℝ → ℝ, (f > 0 ∧ Differentiable ℝ f ∧ ∀ x : ℝ, deriv f x > f x) → (∃ N : ℝ, ∀ x > N, f x > Real.exp (k * x)))
-: allfexN ↔ k ∈ putnam_1994_b3_solution :=
-sorry
+    (S : Set (ℝ → ℝ))
+    (S_def : ∀ f, f ∈ S ↔ (∀ x, 0 < f x ∧ f x < deriv f x) ∧ Differentiable ℝ f) :
+    {k | ∀ f ∈ S, ∃ N, ∀ x > N, Real.exp (k * x) < f x} = putnam_1994_b3_solution := by
```
Comment (eric-wieser) [inline lean4/src/putnam_1994_b3.lean:14]: I'd be tempted to inline `S` here, since it is not named in the question

**Review by eric-wieser — APPROVED** [review]

File: lean4/src/putnam_1994_b3.lean [hunk 2]
```diff
@@ -9,8 +9,7 @@ abbrev putnam_1994_b3_solution : Set ℝ := sorry
...
 -/
 theorem putnam_1994_b3
-(k : ℝ)
-(allfexN : Prop)
-(hallfexN : allfexN = ∀ f : ℝ → ℝ, (f > 0 ∧ Differentiable ℝ f ∧ ∀ x : ℝ, deriv f x > f x) → (∃ N : ℝ, ∀ x > N, f x > Real.exp (k * x)))
-: allfexN ↔ k ∈ putnam_1994_b3_solution :=
-sorry
+    :
+    {k | ∀ f (hf : (∀ x, 0 < f x ∧ f x < deriv f x) ∧ Differentiable ℝ f),
+      ∃ N, ∀ x > N, Real.exp (k * x) < f x} = putnam_1994_b3_solution := by
```
Comment (GeorgeTsoukalas) [inline lean4/src/putnam_1994_b3.lean:14]: Could you remove `by` here? The convention for the rest of the Lean formalizations is not to enter tactic mode by default.

Comment (eric-wieser) [reply lean4/src/putnam_1994_b3.lean:14]: Applied throughout in #239

Comment (ocfnash) [reply lean4/src/putnam_1994_b3.lean]: Done here too (notwithstanding #239).

**Review by GeorgeTsoukalas — APPROVED** [review]
Thanks for the contribution Oliver!

**Pull request merged**